lib/ansible/modules/cloud/kubevirt/kubevirt_vm_status.py

```diff
@@ -132,127 +132,94 @@
-        """ Build API client based on user's configuration """
-        host = self.params.get('host')
-        username = self.params.get('username')
-        password = self.params.get('password')
-        api_key = self.params.get('api_key')
+        wait = self.params.get('wait')
+        wait_time = self.params.get('wait_timeout')
+        existing = None
+        existing_running = None
+        return_attributes = dict(changed=False, result=dict())
 
-        if (host and username and password) or (api_key and host):
-            return self.__configure_by_params()
-        return self.__configure_by_file()
-
-    def __configure_by_file(self):
-        """ Return API client from configuration file """
-        if not self.params.get('kubeconfig'):
-            config_file = os.path.expanduser(
-                kube_config.KUBE_CONFIG_DEFAULT_LOCATION)
-        else:
-            config_file = self.params.get('kubeconfig')
+        resource = self.find_resource(kind, api_version, fail=True)
 
         try:
-            kube_config.load_kube_config(
-                config_file=config_file,
-                context=self.params.get('context')
-            )
-            config = kubernetes.client.Configuration()
-            if not self.params.get('verify_ssl'):
-                setattr(config, 'verify_ssl', False)
-            return kubernetes.client.ApiClient(
-                configuration=config
-            )
-        except (IOError, ConfigException):
-            raise
-
-    def __configure_by_params(self):
-        """ Return API client from configuration file """
-        kube_config.load_kube_config()
-        config = kubernetes.client.Configuration()
-        auth_args = AUTH_ARG_SPEC.keys()
-
-        for key, value in iteritems(self.params):
-            if key in auth_args and value is not None:
-                if key == 'api_key':
-                    setattr(
-                        config,
-                        key, {'authorization': "Bearer {0}".format(value)})
-                else:
-                    setattr(config, key, value)
-
-        if not self.params.get('verify_ssl'):
-            setattr(config, 'verify_ssl', False)
-
-        return kubernetes.client.ApiClient(configuration=config)
-
+            existing = resource.get(name=name, namespace=namespace)
+            return_attributes['result'] = existing.to_dict()
+        except KubernetesException as exc:
+            self.fail_json(msg='Failed to retrieve requested object: {0}'.format(exc.message),
+                           error=exc.value.get('status'))
+
+        if hasattr(existing.spec, 'running'):
+            existing_running = existing.spec.running
+
+        if resource_version and resource_version != existing.metadata.resourceVersion:
+            self.exit_json(**return_attributes)
+
+        # TODO: Not nice code, but working, just not to break, what have been done,
+        # but really need rewrite.
+        if state == 'running':
```

Comment:
What about other states like `Pending` or `Scheduling`?

Reply:
That's something we need to do. Currently the modules has same functionality as before. So we don't break it. The improvement patch can be done later.

Reply:
OK